### PR TITLE
added connection and gateway apis

### PR DESCRIPTION
**Pull request by m-kovalsky**

### CODE REVIEW

File: src/sempy_labs/_connections.py [hunk 1]
```diff
@@ -143,49 +234,93 @@ def list_item_connections(
...
+
+    responses = pagination(client, response)
+
+    records = []
+    for r in responses:
+        for v in r.get("value", []):
+            records.append(
+                {
+                    "Connection Type": v.get("type"),
+                    "Creation Method": v["creationMethods"][0]["name"],
```
Comment (eisber) [inline src/sempy_labs/_connections.py:270]: why not get()?

File: src/sempy_labs/_connections.py [hunk 2]
```diff
@@ -194,157 +329,108 @@ def create_connection_cloud(
...
-    df = pd.concat([df, pd.DataFrame(new_data, index=[0])], ignore_index=True)
-
-    df["Skip Test Connection"] = df["Skip Test Connection"].astype(bool)
 
-    return df
+    print(f"{icons.green_dot} The '{name}' on-prem connection has been created.")
 
 
-def create_connection_vnet(
+def create_vnet_connection(
```
Comment (eisber) [inline src/sempy_labs/_connections.py:395]: are these public functions? if so, docstring?

Comment (m-kovalsky) [reply src/sempy_labs/_connections.py:395]: they weren't public as they weren't in the init file

Comment (m-kovalsky) [reply src/sempy_labs/_connections.py:480]: done

File: src/sempy_labs/_connections.py (hunk 1 above)
Comment (m-kovalsky) [reply src/sempy_labs/_connections.py:307]: true. making this a private function for now as it's not fully ready.

**Review by eisber — APPROVED** [review]

**Pull request merged**